asset_stream_manager/main.cc

```diff
@@ -160,21 +167,24 @@ ABSL_FLAG(std::string, dev_mount_dir, "",
 int main(int argc, char* argv[]) {
   absl::ParseCommandLine(argc, argv);
 
-  // Set up config. Allow overriding this config with
-  // %config_dir%\assets_stream_manager.json.
-  const std::string config_path = cdc_ft::path::Join(
-      absl::GetFlag(FLAGS_config_dir), "assets_stream_manager.json");
+  // Set up config. Allow overriding this config |config_file|.
   cdc_ft::AssetStreamConfig cfg;
-  absl::Status cfg_load_status = cfg.LoadFromFile(config_path);
+  std::string config_file = absl::GetFlag(FLAGS_config_file);
+  absl::Status cfg_load_status =
+      cdc_ft::path::ExpandPathVariables(&config_file);
```

Comment:
Errors from ExpandPathVariables should probably be critical. The user will want to know if they have messed up the filename.

Reply:
A few lines below it is logged, after the logging is initialized.

Reply:
No I mean this should be fatal. We should exit with an error code instead of proceeding. It is definitely something the user will want to know about. If the user misses the error, e.g. if the manager was started in the background or logs to a file, which is the default, it could lead to subtle connection issues, and it would save the user time to understand that the config wasn't loaded.

Reply:
In case of a misspelling we don't know whether the config just doesn't exist or whether it's mistyped. If ExpandPathVariables could tell us when there is a definite error, it would be preferrable to exit at this point. However, it turns out that ExpandPathVariables won't tell us if a variable is mistyped:
"If the name is not found, the %variableName% portion is left unexpanded."
Therefore, there's nothing to gain. Sorry for the fuzz, I should have checked this before.